src/core/lib/security/transport/security_connector.c

```diff
@@ -509,7 +556,8 @@ static grpc_error *ssl_check_peer(grpc_security_connector *sc,
 static void ssl_channel_check_peer(grpc_exec_ctx *exec_ctx,
```

Comment:
Yes indeed.

For the connection to the SSL load balancer, we will need to verify that the load balancer name is on the certificate of the load balancer connection. Right now, if my understanding is correct, SSL will just verify the original target name in the security connector which is empty string for the load balancing subchannel: as a result this subchannel will fail.
Does that make sense?